app/src/main/java/com/example/next_show/data/ShowFilter.java

```diff
@@ -0,0 +1,169 @@
+
+    // filters for network
+    public static final String CABLE = "cable";
+    public static final String STREAMING = "streaming";
+    public static final List<String> STREAMING_SERVICES =
+            Arrays.asList("Netflix", "Hulu", "Disney+", "HBOMax", "Peacock", "Amazon", "Apple TV+");
+
+    // filters for year
+    public static final String THIS_YEAR = "2021";
+    public static final String PAST_FIVE = "2016 - 2020";
+    public static final String PRIOR = "< 2015";
+
+    public static final int CURRENT_YEAR = 2021;
+    public static final int PAST_YEAR = 2020;
+    public static final int PRIOR_YEAR = 2015;
+
+    // filters
+    public static final String GENRE = "genre";
+    public static final String NETWORK = "network";
+    public static final String YEAR = "year"; // year aired!
+
+    public ShowFilter(List<Show> showsList) {
+        // when new Filterer is init, set it up
+        this.showsList = showsList;
+        setUpFilters();
+    }
+
+    public void setUpFilters() {
+        filters = new HashMap<>();
+
+        // initialize dictionary with keys and blank lists
+        filters.put(GENRE, new ArrayList<>());
+        filters.put(NETWORK, new ArrayList<>());
+        filters.put(YEAR, new ArrayList<>());
+
+        // clear shows
+        showsList.clear();
+    }
+
+    public HashMap<String, List<String>> getFilters(){
+        return filters;
+    }
+
+    public void addToGenre(String g) {
+        filters.get(GENRE).add(g);
+    }
+
+    public void addToNetwork(String n) {
+        filters.get(NETWORK).add(n);
+    }
+
+    public void addToYear(String y) {
+        filters.get(YEAR).add(y);
+    }
+
+    public void removeFromGenre(String g) {
+        filters.get(GENRE).remove(g);
+    }
+
+    public void removeFromNetwork(String n) {
+        filters.get(NETWORK).remove(n);
+    }
+
+    public void removeFromYear(String y) {
+        filters.get(YEAR).remove(y);
+    }
+
+    public void updateShowsList(List<Show> newShows) {
+        showsList.addAll(newShows);
+    }
+
+    public void setShowsList(List<Show> newShows) {
+        allShows = newShows;
+    }
+
+    public List<Show> getAllShows() {
+        return allShows;
+    }
+
+    public List<Show> filterShows(List<Show> shows) {
+        List<Show> filteredShows = shows;
+
+        // filter by network (if appplicable)
+        List<String> networkQueries = filters.get(NETWORK);
+        filteredShows = filterNetwork(filteredShows, networkQueries);
+
+        // filter by genre (if appplicable)
+        List<String> genreQueries = filters.get(GENRE);
+        filteredShows = filterGenre(filteredShows, genreQueries);
+
+        // filter by year (if appplicable)
+        List<String> yearQueries = filters.get(YEAR);
+        filteredShows = filterYear(filteredShows, yearQueries);
+
+        return filteredShows;
+    }
+
+    private List<Show> filterYear(List<Show> current, List<String> queries) {
```

Comment:
nit: for all of these -- queries is a little generic.  Can we instead name is something like yearFilters?